relay-redis/src/pool.rs

```diff
@@ -15,6 +17,88 @@ pub type CustomClusterPool = Pool<CustomClusterManager, CustomClusterConnection>
 /// A connection pool for single Redis instance deployments.
 pub type CustomSinglePool = Pool<CustomSingleManager, CustomSingleConnection>;
 
+/// A wrapper for a connection that can be tracked with metadata.
+pub struct TrackedConnection<C> {
+    connection: C,
+    detach: bool,
```

Comment:
I decided to use a `bool` instead of storing the `RedisError` since it does not allow cloning.